### PR TITLE
allow explicit ints in `dynamic_shapes`

### PR DESCRIPTION
**Pull request by BoyuanFeng**
Summary:
`dynamic_shapes` requires `None`s for dimensions with static shapes, which will be decided during compilation time as the same shape as the example inputs given to `torch.export.export()` call.

However, users may want to specify a static shape regardless of the example inputs.

This task enables explicitly specifying `int`s in `dynamic_shapes`.

Differential Revision: D52771777

### CODE REVIEW

File: torch/export/dynamic_shapes.py [hunk 1]
```diff
@@ -420,31 +421,28 @@ def check_same_bounds(dim):
         else:
             bounds[dim.__name__] = (dim.min, dim.max)
 
+    def update_single_symbol(dim, tensor, i):
+        if isinstance(dim, int):
+            dim = Dim(name=f"explicit_static_dim_{num_explicit_static_dims[0]}", min=dim, max=dim)
```
Comment (avikchaudhuri) [inline torch/export/dynamic_shapes.py:426]: Interesting approach. While this should work, I'd prefer some kind of subclass of `Dim` here.

Also, the name has to be unique, there is no guarantee that the name here will not conflict.

File: torch/export/dynamic_shapes.py [hunk 2]
```diff
@@ -55,7 +55,7 @@ def Dim(name: str, *, min: Optional[int] = None, max: Optional[int] = None):
     """
     _min = 2 if min is None else builtins.max(min, 2)
     _max = sys.maxsize - 1 if max is None else builtins.min(max, sys.maxsize - 1)
-    assert _max > _min, f"Cannot create Dim with inconsistent min={min}, max={max}"
+    assert _max >= _min, f"Cannot create Dim with inconsistent min={min}, max={max}"
```
Comment (avikchaudhuri) [inline torch/export/dynamic_shapes.py:58]: Hmm, this might cause problems down the line because everywhere else in symbolic shapes we have special handling of "symbol is specialized to int" than encoding it as "symbol has same upper and lower bound." See below.

**Review by avikchaudhuri — COMMENTED** [review]
Suggested a change to the approach, can you try it and report back?
